### PR TITLE
allow nunjucks configuration

### PR DESCRIPTION
**Pull request by ashleydw**
Enables custom nunjucks configurations to be used, for example:

```
const config = raml2html.getDefaultConfig('./templates/template.nunjucks');

config.setupNunjucks = function(nunjucks) {
  nunjucks.addFilter('stringify', function(object) {
    if(typeof object == 'string')
      return object;
    return JSON.stringify(object, null, 2);
  })
  return nunjucks;
};```

### CODE REVIEW

**Comment by kevinrenskers** [comment]
You can already give your own custom `processRamlObj` function in which Nunjucks is set up. So I am not really sure how this helps?  Yes, it allow you to add Nunjucks specific stuff without overriding the entire `processRamlObj` function, but then where does it end? We'll end up with hooks for everything :)

File: index.js [hunk 1]
```diff
@@ -56,13 +56,18 @@ function getDefaultConfig(mainTemplate, templatesPath) {
...
+      
       renderer.table = function (thead, tbody) {
         // Render Bootstrap style tables
         return `<table class="table"><thead>${thead}</thead><tbody>${tbody}</tbody></table>`;
       };
 
       // Setup the Nunjucks environment with the markdown parser
       const env = nunjucks.configure(templatesPath, { autoescape: false });
+      
+      if(config.setupNunjucks)
```
Comment (kevinrenskers) [inline index.js]: You're not passing the `config` parameter into the `processRamlObj` function so this won't work. It also absolutely does not follow the code style of the project and this won't lint.

Comment (ashleydw) [inline index.js]: forgot to commit the config, added it now

**Comment by ashleydw** [comment]
I understand you can override the `processRamlObj` but overriding the entire thing is overkill for me; i don't want to have to maintain the function in my project, only my extensions

File: index.js (hunk 1 above)
Comment (kevinrenskers) [inline index.js]: Please follow the code style of the project and run `npm run lint`. Thanks!

**Comment by ashleydw** [comment]
Updated to fix lint errors 👍

**Pull request merged**